studio/pages/project/_/[[...routeSlug]].tsx

```diff
@@ -33,25 +33,29 @@ const Header = () => {
 const GenericProjectPage: NextPage = () => {
   const router = useRouter()
   const { routeSlug, ...queryParams } = router.query
-  const queryString =
-    Object.keys(queryParams).length > 0
-      ? new URLSearchParams(queryParams as Record<string, string>).toString()
-      : ''
+  const queryString = new URLSearchParams(queryParams as Record<string, string>)
 
   const urlRewriterFactory = (slug: string | string[] | undefined) => {
     return (projectRef: string) => {
       const hash = location.hash
 
       if (!Array.isArray(slug)) {
-        return [`/project/${projectRef}`, queryString ?? `?${queryString}`, hash ?? `#${hash}`]
+        return [
+          `/project/${projectRef}`,
+          // @ts-ignore [Joshen] https://github.com/microsoft/TypeScript/issues/54466
```

Comment:
Oh, sad :(

Looking at it again, this is same query for all urls, so you can also basically revert to updated initial version with:
```js
const queryString = Object.keys(queryParams).length
    ? `?${new URLSearchParams(queryParams as Record<string, string>)}`
    : undefined;
```

and use `queryString` as value directly, without any checks there. I'm fine with any.

You can also add:
`const hash = location.hash ? `#${location.hash}` : undefined` (or '' if desired).

and make it look pretty with `return [`/project/${projectRef}`, queryString, hash]`